### PR TITLE
Common: Try to improve MSP doc especially for HDZero

### PR DESCRIPTION
**Pull request by timtuxworth**
While setting up my HDZero goggles, I found some of the wiki relating to MSP could be clearer. This is my attempt to improve it.

I've also re-structure the page, there was a lot of general stuff relating to OSD comments that was buried under the DJI Goggles section, so I've pushed the different google setups to the top and then made a common section below for all the stuff that applies regardless which goggles you have.

Also clarified how you use the new Mission Planner "HD Layout" OSD feature.

### CODE REVIEW

File: common/source/docs/common-msp-osd-overview-4.2.rst [hunk 1]
```diff
@@ -41,16 +41,138 @@ Features
...
+To enable telemetry based MSP OSD, set the following parameters ( example using SERIAL port 2 as the port to attach to the DJI Air unit using both TX and RX lines):
 
  - :ref:`OSD_TYPE<OSD_TYPE>` = 3 if no integrated OSD is being used in order to activate the OSD code. If an integrated OSD is present and the user wishes to have both OSDs , then :ref:`OSD_TYPE<OSD_TYPE>` = 1 will activate the on-board OSD as well as providing screens for the MSP OSD function. For example, on vehicles using the DJI goggles/air system for medium range, but still running a long range VTX using the internal OSD for when the vehicle exceeds the range of the HD DJI Goggles. This configuration could use one OSD screen optimized for DJI Goggles, and another for the integrated OSD and the user can switch between them depending on which video system is being viewed.
  - :ref:`SERIAL2_PROTOCOL<SERIAL2_PROTOCOL>` = 33
- - :ref:`MSP_OPTIONS<MSP_OPTIONS>` = 0 (polling mode)
+ - :ref:`SERIAL2_BAUD<SERIAL2_BAUD>` = 115
+ - :ref:`MSP_OPTIONS<MSP_OPTIONS>` bit 0 = 1 (EnableTelemetryMode)
```
Comment (Hwurzburg) [inline common/source/docs/common-msp-osd-overview-4.2.rst:49]: this is incorrect....this is push mode for MSP OSDs using only the TX line not dji goggles

File: common/source/docs/common-msp-osd-overview-4.2.rst [hunk 2]
```diff
@@ -41,16 +41,138 @@ Features
...
  
 .. note:: Serial port buad rate default is changed to 115.2Kbaud automatically when setting the above protocol type. However, if the user has previously or subsequently changes the baud, this default will not be used. 115.2Kbaud is required by most video goggle systems.
 
 .. note:: DJI Custom OSD must be enabled: in SETTINGS->DISPLAY->CUSTOM OSD menu of goggles
 
+DisplayPort OSD
```
Comment (Hwurzburg) [inline common/source/docs/common-msp-osd-overview-4.2.rst:55]: Moving all of this before the panel items section is wrong...these are MSP only panel items...DisplayPort and integrated OSDs display MANY more panels (all of them that are not MSP only)....

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
just quick first pass...will take some more time this weekend

File: common/source/docs/common-msp-osd-overview-4.2.rst (hunk 2 above)
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:55]: Hmm -It is VERY confusing to have all these OSD values listed on the page and have to scroll past all this stuff to get to the DisplayPort part. A couple of options, what do you think:
1. Put DisplayPort first - then the OSD list can be at the bottom of the page and only those with a telemetry OSD have to scroll there to see it
2. Split this into two pages.

File: common/source/docs/common-msp-osd-overview-4.2.rst (hunk 1 above)
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: Ok - I have to say I don't really understand the MSP_OPTIONS bit 1 then, all I know is that if I set it to 1 my HDZero goggles don't work. So when would you ever set it to 1?

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: on an msp osd that is connected only by its rx input to the uarts tx output...normally the osd requests that info be semt..this option allows it to be 'pushed' to the osd

File: common/source/docs/common-msp-osd-overview-4.2.rst (hunk 2 above)
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:55]: Thanks @Hwurzburg - I'll try that.

File: common/source/docs/common-msp-osd-overview-4.2.rst (hunk 1 above)
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: So the bit itself "EnableTelemetryMode" - should actually be "EnableTelemetryPollingMode" ?

Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: > this is incorrect....this is push mode for MSP OSDs using only the TX line not dji goggles

So which goggles would this apply to?

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: a standalone osd like mwosd

Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: So if I say this: 
.. note:: a standalone osd like MWOSD, with only one wire connected, will need :ref:`MSP_OPTIONS<MSP_OPTIONS>` bit 0 = 1 (EnableTelemetryMode). This is the only case where this should be used.

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: I am working on a rewrite to make the various osd types clearer..standby

**Comment by timtuxworth** [comment]
Made some major changes based on your feedback. I'm thinking it flows a lot better now - try reading it "from the top" and see if it makes sense and if I've broken anything.

**Comment by Hwurzburg** [comment]
added info on non dji msp....hopefully this is clear now

**Comment by timtuxworth** [comment]
> added info on non dji msp....hopefully this is clear now

This is definitely an improvement. I think overall much clearer and easier to understand and also easier to find what you need for your specific goggles. I'll do a QA pass probably this afternoon.

File: common/source/docs/common-msp-osd-overview-4.2.rst [hunk 3]
```diff
@@ -3,56 +3,31 @@
...
-.. note:: Serial port buad rate default is changed to 115.2Kbaud automatically when setting the above protocol type. However, if the user has previously or subsequently changes the baud, this default will not be used. 115.2Kbaud is required by most video goggle systems.
+ - :ref:`SERIAL2_PROTOCOL<SERIAL2_PROTOCOL>` = 32 (MSP telemetry only mode)
+ - :ref:`SERIAL2_BAUD<SERIAL2_BAUD>` = 115
+ - :ref:`MSP_OPTIONS<MSP_OPTIONS>` bit 0 = 1 (EnableTelemetryMode), but only if the OSD is to be connected only via a single wire from its RX input to the Serial ports TX. This is known as "push" mode.
 
-.. note:: DJI Custom OSD must be enabled: in SETTINGS->DISPLAY->CUSTOM OSD menu of goggles
-
-OSD Panel Items
----------------
+ - These are the MSP only OSD elements supported on telemetry based OSDs (assuming the OSD has the capability itself).
```
Comment (timtuxworth) [inline common/source/docs/common-msp-osd-overview-4.2.rst:30]: I disagree strongly with you about embedding this table here @Hwurzburg . Sure it makes sense if you have this type of goggles, but for anyone else it makes the page very hard to read and find what you are looking for. It's your call of course.

File: common/source/docs/common-msp-osd-overview-4.2.rst [hunk 4]
```diff
@@ -115,112 +90,157 @@ OSD Panel Items
...
-DisplayPort OSD
-===============
+.. note:: the direction arrows will be reversed since ArduPilot and Betaflight use direction arrows in their font tables that are 180 deg different than each other. This can be corrected by using :ref:`OSD_OPTIONS<OSD_OPTIONS>` bit 5 to invert them before sending to OSD.
 
-HDZero using ArduPilot custom fonts
+MWOSD, Walksnail, and DJI goggles using the wtf-os/msp-osd firmware already support custom fonts locally, and HDZero has an ArduPilot compatible font set.
 
-.. image:: ../../../images/msp_osd_displayport.jpg
-   :target: ../_images/msp_osd_displayport.jpg
+Default behavior (:ref:`MSP_OPTIONS<MSP_OPTIONS>` Bit 2 = 0) is to use the ArduPilot font table's indexes.
```
Comment (timtuxworth) [inline common/source/docs/common-msp-osd-overview-4.2.rst:136]: This discussion of fonts now reads like it only applies to DisplayPort. I'm going to suggest a change which moves it down into the OSD section.

File: common/source/docs/common-msp-osd-overview-4.2.rst [hunk 5]
```diff
@@ -115,112 +90,157 @@ OSD Panel Items
...
 Display Resolution
 ------------------
-
 If the OSD is capable, you can select to display either the SD or HD fonts using ``OSDx_TXT_RES`` for each OSD screen enabled. 0 = SD (30x16), 1 = HD (50x18), 3 = HD (60x22).
 
-The SD font's positions are set on a 30x16 X/Y position grid as normal, the HD uses a 50x18 or 60x22 grid. The 50x18 grid has margins at the top/bottom/left/right of the screen before the grid begins. 
+For HDZero you should set ``OSDx_TXT_RES`` to 0 or 1 for each enabled OSD screen. If you set it to 2, the text displayed will be garbled.
 
-.. note:: Mission Planners' OSD setup screen now suuports HD OSD configuration.
+The SD font's positions are set on a 30x16 X/Y position grid as normal, the HD uses a (1) 50x18 or (2) 60x22 grid. The 50x18 grid has margins at the top/bottom/left/right of the screen before the grid begins.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:208]: TXT _RES only applies to DisplayPort systems with their own fonts and display resolutions...its merely a value shipped to the Display port device

**Comment by timtuxworth** [comment]
The new version reads like the ability to have two OSDs (integrated + Air Unit) does not apply to DisplayPort. As written you can enable Telemetry based OSD + internal or DJI Custom OSD + internal, but NOT DisplayPort + internal.
Is that correct @Hwurzburg ?

File: common/source/docs/common-msp-osd-overview-4.2.rst (hunk 3 above)
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:37]: it belongs with the MSP only stuff

**Comment by Hwurzburg** [comment]
not unless you use the second osd type feature added in 4.5
I must say that the last changes you made (aside from some minor format corrections) I find unacceptable...and I dont have time to individually work on them any more right now....so we will just put this on hold for now

**Comment by timtuxworth** [comment]
> not unless you use the second osd type feature added in 4.5
> I must say that the last changes you made (aside from some minor format corrections) I find unacceptable...and I dont have time to individually work on them any more right now....so we will just put this on hold for now

Totally get it Henry - you have a lot on, I don't know how you do it. I'm also working on a video just around ArduPilot with HDZero, so no rush.

**Comment by Hwurzburg** [comment]
I have included a lot of your changes from the previous push that  I commented on...I am merging this without the later pushes..please open a new PR if you want to make additional changes to make review easier..

**Comment by Hwurzburg** [comment]
@timtuxworth I had a git meltdown and lost all of my changes!! going to take a break and restart later...sorry for the FU

**Comment by timtuxworth** [comment]
Thanks for working with me on this @Hwurzburg . I'm pretty sure we've made this a lot easier to follow and to understand how to set up any specific set of goggles. I'm going to do some flying next! :-)

**Pull request merged**